### DISCUSSION TITLE
Can't load SQLite extension (e.g asg017/regex)

### DISCUSSION THREAD

**Discussion by bobhy** [dbeaver/dbeaver#34531] [Q&A] [answered]
### Description

(I'm new to DBeaver, be warned)
I cannot get DBeaver(-ce) to load a SQLite extension to enable regexp functions.  First blocker is I do not see the 'extensions' tab in connection settings panel, as is shown in the help doc.  I have also tried configuring the extension 'libraries' tab in the (sqlite) driver settings panel, but this doesn't work and I'm not sure that's even correct for a native dynamic library .so file.  
I *can* load the extension in sqlite3 invoked from the command line and successfully run `select 'abc' regexp 'b'`, so there's that.

### DBeaver Version

Community Edition Version 24.1.1.202406231636

### Operating System

Ubuntu noble

### Database and driver

From help / installation details:
*** Date: Tuesday, June 25, 2024 at 6:24:17 PM Eastern Daylight Time

*** Platform Details:

*** System properties:
DRIVERS_LOCATION=/home/bobhy/snap/dbeaver-ce/309/.local/share/DBeaverData/drivers
applicationXMI=org.eclipse.ui.workbench/LegacyIDE.e4xmi
dbeaver.distribution.type=deb
dbeaver.io.tmpdir=/home/bobhy/Downloads
ds.delayed.keepInstances=true
ds.delayed.keepInstances.default=true
eclipse.application=org.jkiss.dbeaver.ui.app.standalone.standalone
. . .

<pre><font color="#39BAE6">$ cat</font><font color="#B3B1AD"> </font><font color="#B3B1AD"><u style="text-decoration-style:solid">configuration.txt</u></font><font color="#B3B1AD"> </font><font color="#F29668">|</font><font color="#B3B1AD"> </font><font color="#39BAE6">grep</font><font color="#B3B1AD"> sqlite</font>
org.jkiss.dbeaver.ext.<font color="#C01C28"><b>sqlite</b></font> (1.0.182.202406231636) &quot;DBeaver SQLite Model&quot; [Active]
org.jkiss.dbeaver.ext.<font color="#C01C28"><b>sqlite</b></font>.ui (1.0.103.202406231636) &quot;DBeaver SQLite UI&quot; [Starting]
Id: org.jkiss.dbeaver.ext.<font color="#C01C28"><b>sqlite</b></font>, Version: 1.0.182.202406231636, Location: reference:file:plugins/org.jkiss.dbeaver.ext.<font color="#C01C28"><b>sqlite</b></font>_1.0.182.202406231636.jar
Id: org.jkiss.dbeaver.ext.<font color="#C01C28"><b>sqlite</b></font>.ui, Version: 1.0.103.202406231636, Location: reference:file:plugins/org.jkiss.dbeaver.ext.<font color="#C01C28"><b>sqlite</b></font>.ui_1.0.103.202406231636.jar
</pre>

### Steps to reproduce

1. Open DBeaver-ce; new db connection
2. in Connect to a database form, observe no 'extensions' tab, only 'main' and 'driver properties'.  [Help doc](https://dbeaver.com/docs/dbeaver/Database-driver-SQLite/) shows it should be there.

![image](https://github.com/dbeaver/dbeaver/assets/1971300/449b57ac-18d0-4711-8a43-73cf35e1f597)

3. Undaunted, next click on 'Driver Settings', move to 'libraries' tab, click 'add file' and select the extension `.so` from `/home/<me>/.sqlpkg/asg017/regex/`.  Note it gets uploaded to a different path (which changes over time and is not cached by `ldconfig`...)

![image](https://github.com/dbeaver/dbeaver/assets/1971300/d2b22f9a-1057-4de4-860e-ce1fb298924c)  

### Additional context

I could easily be trying the wrong thing.  Is there another way to get DBeaver to load this sqlite extension when I open the db connection?

**Comment by ShadelessFox** [comment, marked as answer]
Hello @bobhy,

Extension management is a commercial feature of all our PRO products. We will update the wiki to state it explicitly.

You can get a trial version here: https://dbeaver.com/download/

Otherwise, you can do the following:
1. Set the `enable_load_extension` property to `true` in **Driver properties**
2. Execute `SELECT load_extension('extension-name.so')` either as a bootstrap query or in the SQL editor

Remember that you need to have a 64-bit version of the extension library. Please let me know if it helps

**Comment by bobhy** [comment]
Thanks!  That worked for me, closing the issue.

I would be happy to write up a blurb for the wiki describing this for users of the  community edition...

**Reply by bobhy** [reply]
Another thing I'm going to try is to run sqlite an external process, then configure DBeaver to access it "remotely" via SSH.  Since DBeaver is relying on JDBC, I don't think I'd lose any DBeaver capabilities, though maybe a bit of performance.

The advantage of running sqlite externally is I could use standard sqlite tools and doc to load the extension, or could choose to run a prebuilt sqlite3 executable that has extensions built in (such as https://github.com/nalgeon/sqlean).

**Reply by O-J1** [reply]
Did you ever get this working? I could not (likely a skill issue on my part)

**Reply by bobhy** [reply]
@O-J1 I never pushed further into running SQLITE via SSH, the answer above was all I needed.

The community edition allows you to run a script on initial connect to the DB (Connection Settings / Shell Commands / Event: After Connect), so you could but the `SELECT load_extension()` there, but I found it easier to just put it at the top of the SQL scripts I was already using.
